### ISSUE TITLE
[Object-detection] the error message of \src\github\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:1108] could not synchronize on CUDA context: CUDA_ERROR_LAUNCH_FAILED ::

### ISSUE THREAD

**Issue by wenouyang** [tensorflow/models#5328]
When training the Google object-detection model, the training process was broken with the following error message, what can be the underlying reason?

```
INFO:tensorflow:global step 16171: loss = 2.1875 (1.656 sec/step)
INFO:tensorflow:global step 16171: loss = 2.1875 (1.656 sec/step)
INFO:tensorflow:global step 16172: loss = 2.5936 (1.560 sec/step)
INFO:tensorflow:global step 16172: loss = 2.5936 (1.560 sec/step)
2018-09-18 15:21:34.702531: E T:\src\github\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:1108] could not synchronize on CUDA context: CUDA_ERROR_LAUNCH_FAILED ::
2018-09-18 15:21:34.702567: E T:\src\github\tensorflow\tensorflow\stream_executor\cuda\cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_LAUNCH_FAILED
2018-09-18 15:21:34.712913: F T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_event_mgr.cc:206] Unexpected Event status: 1


```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by amin07** [comment]
am also having the same problem with cuda 9.0, cudnn 7.1 and tensorflow 1.10
Please help..
.
.
.
.
INFO:tensorflow:global step 4197: loss = 1.5479 (0.700 sec/step)
INFO:tensorflow:global step 4197: loss = 1.5479 (0.700 sec/step)
INFO:tensorflow:global step 4198: loss = 1.3282 (0.695 sec/step)
INFO:tensorflow:global step 4198: loss = 1.3282 (0.695 sec/step)
INFO:tensorflow:global step 4199: loss = 1.1196 (0.690 sec/step)
INFO:tensorflow:global step 4199: loss = 1.1196 (0.690 sec/step)
INFO:tensorflow:global step 4200: loss = 1.6031 (0.701 sec/step)
INFO:tensorflow:global step 4200: loss = 1.6031 (0.701 sec/step)
INFO:tensorflow:global step 4201: loss = 0.8924 (0.697 sec/step)
INFO:tensorflow:global step 4201: loss = 0.8924 (0.697 sec/step)
INFO:tensorflow:global step 4202: loss = 0.8030 (0.688 sec/step)
INFO:tensorflow:global step 4202: loss = 0.8030 (0.688 sec/step)
INFO:tensorflow:global step 4203: loss = 0.9386 (0.697 sec/step)
INFO:tensorflow:global step 4203: loss = 0.9386 (0.697 sec/step)
INFO:tensorflow:global step 4204: loss = 2.0923 (0.692 sec/step)
INFO:tensorflow:global step 4204: loss = 2.0923 (0.692 sec/step)
INFO:tensorflow:global step 4205: loss = 0.9616 (0.678 sec/step)
INFO:tensorflow:global step 4205: loss = 0.9616 (0.678 sec/step)
INFO:tensorflow:global step 4206: loss = 0.9810 (0.684 sec/step)
INFO:tensorflow:global step 4206: loss = 0.9810 (0.684 sec/step)
INFO:tensorflow:global step 4207: loss = 0.9181 (0.687 sec/step)
INFO:tensorflow:global step 4207: loss = 0.9181 (0.687 sec/step)
INFO:tensorflow:global step 4208: loss = 1.2980 (0.703 sec/step)
INFO:tensorflow:global step 4208: loss = 1.2980 (0.703 sec/step)
INFO:tensorflow:global step 4209: loss = 1.1381 (0.682 sec/step)
INFO:tensorflow:global step 4209: loss = 1.1381 (0.682 sec/step)
INFO:tensorflow:global step 4210: loss = 0.9842 (0.682 sec/step)
INFO:tensorflow:global step 4210: loss = 0.9842 (0.682 sec/step)
INFO:tensorflow:global step 4211: loss = 0.9015 (0.671 sec/step)
INFO:tensorflow:global step 4211: loss = 0.9015 (0.671 sec/step)
INFO:tensorflow:global step 4212: loss = 0.8821 (0.693 sec/step)
INFO:tensorflow:global step 4212: loss = 0.8821 (0.693 sec/step)
INFO:tensorflow:global step 4213: loss = 0.6155 (0.690 sec/step)
INFO:tensorflow:global step 4213: loss = 0.6155 (0.690 sec/step)
INFO:tensorflow:global step 4214: loss = 1.0448 (0.691 sec/step)
INFO:tensorflow:global step 4214: loss = 1.0448 (0.691 sec/step)
2018-09-25 20:21:18.428631: E tensorflow/stream_executor/cuda/cuda_driver.cc:1227] failed to enqueue async memcpy from host to device: CUDA_ERROR_LAUNCH_FAILED; GPU dst: 0x120c16ab00; host src: 0x7f31ac400000; size: 4194304=0x400000
2018-09-25 20:21:18.428699: F tensorflow/stream_executor/cuda/cuda_dnn.cc:211] Check failed: status == CUDNN_STATUS_SUCCESS (7 vs. 0)Failed to set cuDNN stream.
2018-09-25 20:21:18.428750: E tensorflow/stream_executor/cuda/cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_LAUNCH_FAILED
Aborted (core dumped)

**Comment by shiyongde** [comment]
ame problem with cuda 9.0, cudnn 7  and tensorflow 1.12.0
NFO:tensorflow:global step 16200: loss = 1.5903 (0.806 sec/step)
INFO:tensorflow:global step 16300: loss = 2.1568 (0.804 sec/step)
2018-12-11 00:31:09.918511: E tensorflow/stream_executor/cuda/cuda_driver.cc:1130] failed to enqueue async memcpy from host to device: CUDA_ERROR_LAUNCH_FAILED: unspecified launch failure; GPU dst: 0x4232ce0b00; host src: 0x7f1f10244780; size: 1597440=0x186000
2018-12-11 00:31:09.918525: E tensorflow/stream_executor/cuda/cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_LAUNCH_FAILED: unspecified launch failure
2018-12-11 00:31:09.918534: E tensorflow/stream_executor/cuda/cuda_driver.cc:1000] could not wait stream on event: CUDA_ERROR_LAUNCH_FAILED: unspecified launch failure
2018-12-11 00:31:09.918534: E tensorflow/stream_executor/cuda/cuda_driver.cc:1000] could not wait stream on event: CUDA_ERROR_LAUNCH_FAILED: unspecified launch failure
2018-12-11 00:31:09.918568: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:274] Unexpected Event status: 1

**Comment by karkirowle** [comment]
I can confirm this is a problem on:
Windows 10
Pre-built 1.10 Tensorflow-GPU
Cuda Compute Capability 3.0 (I know it is not supported)
Cuda 9.0
Multi-GPU model, there wasn't any problem before adding Batch Normalisation (can't share the full model due to IP), but I still need to confirm that.

Occurs after Epoch 20
`E T:\src\github\tensorflow\tensorflow\stream_executor\cuda\cuda_driver.cc:1108] could not synchronize on CUDA context: CUDA_ERROR_ILLEGAL_ADDRESS :: 
2019-07-26 16:09:56.403731: E T:\src\github\tensorflow\tensorflow\stream_executor\cuda\cuda_event.cc:48] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS
2019-07-26 16:09:56.417068: F T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_event_mgr.cc:206] Unexpected Event status: 1`

**Comment by karkirowle** [comment]
On a second note, after checking I don't think BatchNorm is the culprit, and the problem doesn't appear if a train on a smaller subset of the dataset, and same batchsize, so it is very difficult to reproduce. Could I request some help on where could be the problem, what approaches are worth to try?

**Comment by karkirowle** [comment]
By running the neural network on a single GPU, it seems to work, so one possible culprit is the way multi-GPU models are managed. (I use Keras on top of TF to manage multi-gpu training)

**Comment by zheng-yuwei** [comment]
@karkirowle Hi, I encounter the same problem when using tf.keras.utils.multi_gpu_model to train my model on multi-GPUs, while single gpu works fine. Do you find out the reason and solved the problem? thanks ~

tensorflow/stream_executor/cuda/cuda_dnn.cc:503] Check failed: cudnnSetTensorNdDescriptor(handle_.get(), elem_type, nd, dims.data(), strides.data()) == CUDNN_STATUS_SUCCESS (9 vs. 0)batch_descriptor: {count: 1536 feature_map_count: 64 spatial: 208 208 value_min: 0.000000 value_max: 0.000000 layout: BatchDepthYX}
